lib/cocoapods/generator/xcconfig/pod_xcconfig.rb

```diff
@@ -66,6 +70,11 @@ def generate
           end
           XCConfigHelper.add_target_specific_settings(target, @xcconfig)
           @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, target.recursive_dependent_targets)
+          if @test_xcconfig
```

Comment:
For test `xcconfigs` we ensure that linker flags are added for their dependencies.